File: csharp/pascals-triangle/PascalsTriangle.cs
using System;
using System.Collections.Generic;
using System.Linq;

public static class PascalsTriangle
{
    public static IEnumerable<IEnumerable<int>> Calculate(int rows)
    {
        List<IEnumerable<int>> triangle = new List<IEnumerable<int>>();

        while (triangle.Count() < rows)
        {
            yield return buildRow(new List<int>());
        }

    }

    public static IEnumerable<int> buildRow(IEnumerable<int> previousRow)
    {
        yield return 1;

        for (int i = 0; i < previousRow.Count() + 1; i++)
        {
            int n1 = i - 1 >= 0 ? previousRow.ElementAt(i - 1) : 0;
            int n2 = i < previousRow.Count() ? previousRow.ElementAt(i) : 0;
            yield return n1 + n2;
        }
    }

}

File: dart/pascals-triangle/lib/pascals_triangle.dart
class PascalsTriangle {
  List<List<int>> rows(int rowCount) {
    List<List<int>> triangle = [];

    while (triangle.length < rowCount) {
      var nextRow = triangle.length > 0 ? buildRow(triangle.last) : [1];
      triangle.add(nextRow);
    }
    return triangle;
  }

  List<int> buildRow(List<int> previousRow) {
    List<int> newRow = new List<int>(previousRow.length + 1);

    for (int i = 0; i < newRow.length; i++) {
      int n1 = i - 1 >= 0 ? previousRow[i - 1] : 0;
      int n2 = i < previousRow.length ? previousRow[i] : 0;
      newRow[i] = n1 + n2;
    }
    return newRow;
  }
}
